include/cooking/cookingJavascript/searchbar.php

```diff
@@ -0,0 +1,20 @@
+<script>
+function searchBar() {
+    // Declare variables
+    var input, filter, ul, li, a, i;
+    input = document.getElementById('myInput');
+    filter = input.value.toUpperCase();
+    ul = document.getElementById("myUL");
+    li = myUL.getElementsByTagName('li');
+
+    // Loop through all list items, and hide those who don't match the search query
+    for (i = 0; i < li.length; i++) {
+        a = li[i].getElementsByTagName("a")[0];
+        if (a.innerHTML.toUpperCase().indexOf(filter) > -1) {
+            li[i].style.display = "";
```

Comment:
This works (and is impressively clean) but it's probably not how you'd do it in a more advanced web app. That's no problem, your approach is good. But if you're looking for another approach, you could have all the data stored in a js object (rather than just having the whole DOM printed). Then when a search is performed, loop through the elements, find the ones that match, and build a string with the new html which you then insert into the DOM.

The advantage of this approach is it gives you more control over how things are displayed. For example, you might want to sort things differently based on their relevance to the search terms. Right now you can only show/hide things, but with the alternate approach, you can manipulate the data however you want before generating the new html.